### PR TITLE
Fix push handling by removing the PushEvent in favor of Map

### PR DESCRIPTION
**Pull request by drewroengoogle**
https://github.com/flutter/flutter/issues/134746

* The github `Push` event has discrepancies to other similar objects sent and returned by Github, which was causing the automatic conversion from json to fail. Instead of adding more bloat to the github package through adding extensions of objects, use a map since there are only a few values we need from the webhook.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/lib/src/request_handlers/github/webhook_subscription.dart [hunk 1]
```diff
@@ -114,13 +114,13 @@ class GithubWebhookSubscription extends SubscriptionHandler {
         }
         break;
       case 'push':
-        final PushEvent pushEvent = PushEvent.fromJson(json.decode(webhook.payload) as Map<String, dynamic>);
-        final String branch = pushEvent.ref!.split('/')[2]; // Eg: refs/heads/beta would return beta.
-        final String repository = pushEvent.repository!.name;
+        final Map<String, dynamic> event = jsonDecode(webhook.payload) as Map<String, dynamic>;
+        final String branch = event["ref"].split('/')[2]; // Eg: refs/heads/beta would return beta.
```
Comment (ricardoamador) [inline app_dart/lib/src/request_handlers/github/webhook_subscription.dart:118]: Typically we use single quotes. Only use double quotes if your string contains single quotes.

Comment (ricardoamador) [reply app_dart/lib/src/request_handlers/github/webhook_subscription.dart:118]: https://dart.dev/tools/linter-rules/prefer_single_quotes

Comment (drewroengoogle) [reply app_dart/lib/src/request_handlers/github/webhook_subscription.dart]: Thanks! I updated the instances of double quotes in my PR.

**Review by ricardoamador — APPROVED** [review]
LGTM

**Pull request merged**